### PR TITLE
Refactoring: changed user flag enum to enum class

### PR DESCRIPTION
**Pull request by Skeevert**
Static analyzer issued a warning about enum signed integer overflow, so I decided to take a look and discovered a TODO mark that suggests refactoring.

### CODE REVIEW

File: src/discord/user.cpp [hunk 1]
```diff
@@ -221,33 +221,33 @@ void UserData::update_from_json(const nlohmann::json &j) {
 
 const char *UserData::GetFlagName(uint64_t flag) {
     switch (flag) {
```
Comment (ouwou) [inline src/discord/user.cpp:223]: can we just do this so we dont need a wall of static_casts ?
```cpp
    switch (static_cast<UserData::EFlags>(flag)) {
```

Comment (Skeevert) [reply src/discord/user.cpp:223]: For some reason I thought that this cast would be an undefined behavior. Yeah, that's actually much better

Comment (Skeevert) [reply src/discord/user.cpp]: Fixed.

**Comment by Skeevert** [comment]
@ouwou Could you please help with the failed checks?
I don't understand what's wrong with the linker

**Comment by ouwou** [comment]
oh yeah. that has nothing to do with the PR so ill merge anyways and ill fix it on master

**Pull request merged**